spv/sync.go

```diff
@@ -135,21 +149,17 @@ func (s *Syncer) SetNotifications(ntfns *Notifications) {
 
 // synced checks the atomic that controls wallet syncness and if previously
 // unsynced, updates to synced and notifies the callback, if set.
-func (s *Syncer) synced() {
-	if atomic.CompareAndSwapUint32(&s.atomicWalletSynced, 0, 1) &&
-		s.notifications != nil &&
-		s.notifications.Synced != nil {
-		s.notifications.Synced(true)
+func (s *Syncer) synced(walletID int) {
+	if s.notifications != nil && s.notifications.Synced != nil {
```

Comment:
For this `synced` method and the next `unsynced` method, the atomic swap checks were removed. If instead of a single `s.atomicWalletSynced` property, we have a map of `atomicWalletSynced` properties for each wallet, we can retain this functionality.

Instead of `atomicWalletSynced   uint32` in the `Syncer` struct, we can have `atomicWalletsSynced   map[int]*uint32`. Using pointers ensure that when changes from atomic swaps are retained in the map.

Then comparison and swap can be done thus:
`if atomic.CompareAndSwapUint32(s.atomicWalletsSynced[walletID], 0, 1)`